Assets/Character/AaronTestAssets/RunState.cs

```diff
@@ -0,0 +1,91 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class RunState : PlayerState {
+
+    private Rigidbody2D rgb;
+    private Player play;
+    private Animator an;
+
+    void Start () {
+        rgb = GetComponent<Rigidbody2D>();
+        play = GetComponent<Player>();
+        an = GetComponent<Animator>();
+        airborne = GetComponent<AirborneState>();
+        idle = GetComponent<IdleState>();
+        roll = GetComponent<RollState>();
+        attack = GetComponent<AttackState>();
+        backstep = GetComponent<BackstepState>();
+        name = "Run";
+    }
+
+
+
+    public override void ComponentUpdate()
+    {
+        Debug.Log("direction: " + direction + ", speed x: " + rgb.velocity.x);
+
+        if (!an.GetCurrentAnimatorStateInfo(0).IsName("Attack") && !an.GetCurrentAnimatorStateInfo(0).IsName("Jump"))
+        {
+            an.Play("Walk");   
+        }
+
+
+
+        rgb.AddForce(Vector2.right * direction * play.getSpeed(), ForceMode2D.Impulse);
+        Flip();
+        
+
+        if(!grounded)
+        {
+            play.changeState(airborne);
+        }
+        handleInput();
+    }
+
+    public override void handleInput()
+    {
+        if (Input.GetButtonDown("Backstep") && grounded)
+        {
+            backstep.setStartTime(Time.time);
+            play.changeState(backstep);
```

Comment:
If possible, avoid having each state know about the parent Player object. If not possible, I'm not completely against this solution - I think our early SceneManager last year did something similar.